### PR TITLE
bug 1490727: Fix payment submission analytic event and add error event.

### PR DESCRIPTION
**Pull request by joshJarr**
- [x] - Send the submission or error events if a user submits or encounters an error.
- [x] - The amount paid is set in pennies in the event as a value.
- [x] - The location of the submission (banner or FAQ) is also set.
- [x] - Event defaults to no data if nothing can be found.

### CODE REVIEW

File: kuma/settings/common.py [hunk 1]
```diff
@@ -953,8 +953,9 @@ def pipeline_one_scss(slug, **kwargs):
         'source_filenames': (
             'js/payments-handler.js',
             'js/payments-faq.js',
+            'js/payment-confirmation.js',
```
Comment (matthewhall) [inline kuma/settings/common.py]: Should this be "payments-confirmation.js" to match the others?

File: kuma/static/js/payment-confirmation.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                label: 'completed',
+                value: amountSubmitted
+            }, function() {
+                sessionStorage.removeItem(sessionStoreKey);
+            });
+        }
+
+    } else if (path.includes('/payments/error')) {
+        if (amountSubmitted) {
+            mdn.analytics.trackError('Payment error', 'Payment failed');
```
Comment (matthewhall) [inline kuma/static/js/payment-confirmation.js]: Do we also need to remove the `amountSubmitted` from `sessionStorage` here?

File: kuma/static/js/payment-confirmation.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+(function(win) {
+    'use strict';
+
+    // Fire analytic events only if the user has followed payment
+    // flow denoted by `amountSubmitted` session varible.
+    var sessionStoreKey = 'amountSubmitted';
+    var amountSubmitted = sessionStorage.getItem(sessionStoreKey);
+    var path = win.location.pathname;
+    if (path.includes('/payments/success')) {
+        if (amountSubmitted) {
```
Comment (matthewhall) [inline kuma/static/js/payment-confirmation.js]: Can we include this in the `if` statement logic on the line about? `path.includes('/payments/success') && amountSubmitted`.

File: kuma/static/js/payment-confirmation.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                category: 'payments',
+                action: 'submission',
+                label: 'completed',
+                value: amountSubmitted
+            }, function() {
+                sessionStorage.removeItem(sessionStoreKey);
+            });
+        }
+
+    } else if (path.includes('/payments/error')) {
```
Comment (matthewhall) [inline kuma/static/js/payment-confirmation.js]: Same here.

File: kuma/static/js/payments-handler.js [hunk 5]
```diff
@@ -208,13 +208,7 @@
...
                 stripeToken.val(token.id);
                 addDisabledLocaleStorageItem();
-                // Send GA Event.
-                mdn.analytics.trackEvent({
-                    category: 'payments',
-                    action: 'submission',
-                    label: 'completed',
-                    value: selectedAmount * 100
-                });
+                sessionStorage.setItem('amountSubmitted', selectedAmount * 100);
```
Comment (matthewhall) [inline kuma/static/js/payments-handler.js]: How come we're multiplying the amount here?

File: kuma/static/js/payment-confirmation.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+(function(win) {
+    'use strict';
+
+    // Fire analytic events only if the user has followed payment
+    // flow denoted by `amountSubmitted` session varible.
+    var sessionStoreKey = 'amountSubmitted';
```
Comment (matthewhall) [inline kuma/static/js/payment-confirmation.js]: Better to go with something slightly more descriptive: `amountSubmittedStoreKey` or something.

File: kuma/static/js/payments-handler.js (hunk 5 above)
Comment (joshJarr) [reply kuma/static/js/payments-handler.js]: I'll add a comment: but we should convert the amount to pennies to match other events.

**Pull request closed without merging**